### PR TITLE
task(auth): Prevent possibility of double emitting logs

### PR DESCRIPTION
**Pull request by dschom**
## Because

- When two loggers with the same name are created, logs end up being emitted twice.

## This pull request

- Prevents creating loggers with the same name.

## Issue that this pull request solves

Closes: FXA-6031

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/log.js [hunk 1]
```diff
@@ -17,9 +17,19 @@ const Sentry = require('@sentry/node');
 const ISSUER = config.get('domain') || '';
 const CLIENT_ID_TO_SERVICE_NAMES = config.get('oauth.clientIds') || {};
 
+const _registered = {};
+
 function Lug(options) {
+  const name = options.name || 'fxa-auth-server';
+
+  if (_registered[name] != null) {
```
Comment (pdehaan) [inline packages/fxa-auth-server/lib/log.js:25]: I'm always confused by non-strict equals... Should this be a stricter `!== undefined` to avoid possible coercion?

```sh
> const foo = {}
> foo.bar = "dan" # 'dan'
> foo # { bar: 'dan' }
> foo.baz == null # true
> foo.baz == undefined # true
> foo.baz === null # false
> foo.baz === undefined # true
```

Or conversely, a `Set()`, but that might be too heavy:

```sh
> const bar = new Set()
> bar.add("foo") # Set(1) { 'foo' }
> bar.has("foo") # true
> bar.has("baz") # false
```

Comment (dschom) [reply packages/fxa-auth-server/lib/log.js]: The check is logically sound and I don't feel as though coercion is a problem. As your cases indicate `x != null` is equivalent to `x !== null || x !== undefined` and IMO should be seen as idiomatic JavaScript.

I do agree that in many cases a strict equality check is preferable... Since the intent is to flag with `true`, I actually think `if(_registered[name] === true)`, is the most expressive way to write this, so I think I'll switch it to that.

File: packages/fxa-auth-server/lib/log.js [hunk 2]
```diff
@@ -17,22 +17,44 @@ const Sentry = require('@sentry/node');
...
   EventEmitter.call(this);
-  this.name = options.name || 'fxa-auth-server';
+
+  // Ensure that names are unique. If two loggers are created with the same
+  // name, a double log scenario occurs.
+  if (_registered[name] === undefined) {
+    this.name = name;
+  } else {
+    let i = 1;
+    while (_registered[`${name}-${i}`] !== undefined) {
```
Comment (chenba) [inline packages/fxa-auth-server/lib/log.js:33]: nit: Why not keep the latest or next count in ``_registered[`${name}`]``?

File: packages/fxa-auth-server/test/local/log.js [hunk 3]
```diff
@@ -175,6 +175,38 @@ describe('log', () => {
...
+      level: 'debug',
+      name: 'test-duplicates',
+      stdout: { on: sinon.spy() },
+    };
+    logModule(opts);
+    logModule(opts);
+    logModule(opts);
+
+    // Edge case, user passes in already incremented name for some reason...
+    logModule({ ...opts, name: 'test-duplicates-1' });
```
Comment (chenba) [inline packages/fxa-auth-server/test/local/log.js:191]: 😱

**Review by chenba — APPROVED** [review]
Looks good.  I think even terminating the program is justified here since I don't think anyone would intentionally create two logger of the same name.

File: packages/fxa-auth-server/lib/log.js (hunk 2 above)
Comment (dschom) [reply packages/fxa-auth-server/lib/log.js:33]: That'd be a nice touch 👍 .

**Comment by dschom** [comment]
> I think even terminating the program is justified here since I don't think anyone would intentionally create two logger of the same name.

This is what I aimed to do initially. Doing this broke a lot of tests though. I investigated some work a rounds but nothing panned out, so I went with this approach.
